internal/rsl/rsl.go

```diff
@@ -513,6 +514,64 @@ func GetFirstReferenceEntryForRef(repo *git.Repository, targetRef string) (*Refe
 	return firstEntry, annotations, nil
 }
 
+// SkipAllInvalidReferenceEntriesForRef identifies invalid RSL reference entries.
+// Each invalid entry points to a target that is not reachable for the current
+// target of the same reference, indicating that history has been rewritten via a
+// rebase for the reference. After the invalid entries are identified, an annotation
+// entry is created that marks all of these entries as to be skipped.
+func SkipAllInvalidReferenceEntriesForRef(repo *git.Repository, targetRef string, signCommit bool) error {
+	slog.Debug("Checking if RSL entries point to non-existent commits...")
+
+	latestEntry, _, err := GetLatestReferenceEntryForRef(repo, targetRef)
+	iterator := Entry(latestEntry)
+	if err != nil {
+		return err
+	}
+
+	entriesToSkip := []plumbing.Hash{}
+
+	for {
+		if entry, ok := iterator.(*ReferenceEntry); ok {
+			latestEntryCommit, err := gitinterface.GetCommit(repo, latestEntry.TargetID)
+
+			var isAncestor bool
```

Comment:
move this above the GetCommit statement: we don't want anything between assigning `err` and checking it.